### ISSUE TITLE
Bulk registering of reporters

### ISSUE THREAD

**Issue by stevegrunwell** [stevegrunwell/wp404#2]
As more and more useful reporters get built into the plugin it would be nice to have a single filter that accepts an array of callbacks and will hook them all up in one fell swoop.

Ideally we'd be able to return the following array from a `wp404_default_reporters` hook and automatically register them:

``` php
$reporters = array(
  'WP404\Reporters\queries', // Explicitly-named callback
  'post_exists', // Check the WP404\Reporters namespace first, then global
  '\my_function', // Custom function explicitly in the global namespace,
  array( // An array of arguments, used to specify priority as well
    'reporter' => 'server_superglobal',
    'priority' => 20,
  ),
  array( 'class', 'method' ), // Class method
  function ( $report ) { ... }, // Closure
);
```

Once this is in place the plugin default hooks could be registered there instead of in the main plugin file.
